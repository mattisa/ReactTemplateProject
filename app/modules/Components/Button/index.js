import React, { Component } from 'react';
import { observer } from 'mobx-react';
import { observable, action } from 'mobx';

import styles from './styles.css';

@observer
export default class Button extends Component {

    @observable clicked = 'not clicked';

    constructor(props) {
        super(props);
        this._onClick = this._onClick.bind(this);
    }

    @action
    _onClick() {
        this.clicked = 'clicked';
        this.props.onClick();
    }

    render() {
        return (
            <button onClick={this._onClick}>{this.clicked}</button>
        );
    }
}
Button.propTypes = {
    onClick: React.PropTypes.func.isRequired
}

// Button.defaultProps = {
// }
