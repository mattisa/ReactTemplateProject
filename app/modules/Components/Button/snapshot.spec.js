// Link.react-test.js
import React from 'react';
import Button from './index';
import renderer from 'react-test-renderer';

let component;
let tree;
test('Button changes the text when clicked', () => {
    component = renderer.create(
        <Button onClick={() => {}} />
    );

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.onClick();

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
