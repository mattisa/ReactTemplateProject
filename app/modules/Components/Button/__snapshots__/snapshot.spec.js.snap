exports[`test Button changes the text when clicked 1`] = `
<button
  onClick={[Function]}>
  not clicked
</button>
`;

exports[`test Button changes the text when clicked 2`] = `
<button
  onClick={[Function]}>
  clicked
</button>
`;
